pkg/controller/service/service_controller.go

```diff
@@ -705,7 +711,7 @@ func loadBalancerIPsAreEqual(oldService, newService *v1.Service) bool {
 // syncService will sync the Service with the given key if it has had its expectations fulfilled,
 // meaning it did not expect to see any more of its pods created or deleted. This function is not meant to be
 // invoked concurrently with the same key.
-func (s *ServiceController) syncService(key string) error {
+func (s *ServiceController) syncService(ctx context.Context, key string) error {
```

Comment:
Another context-free comment (pun intended), it feels weird to add this to every method, should it be a member variable instead?

Reply:
From the golang docs on context (https://golang.org/pkg/context/) "Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:"